mlops-multi-account-cdk/mlops-sm-project-template/seed_code/deploy_app/deploy_endpoint/batch_inference_pipeline.py

```diff
@@ -0,0 +1,346 @@
+            step_create_model: transform step depends on the create model step
+        Returns:
+            TransformStep
+        '''
+        step_name = "transform"
+        output_path = self.get_output_path(step_name)
+
+        transform = Transformer(
+            model_name=self.model_name,
+            instance_count=1,
+            instance_type=self.inference_instance_type,
+            output_path=output_path,
+            base_transform_job_name=f"{self.pipeline_name}/{step_name}",
+            max_payload=10,
+            accept='text/csv'
+        )
+
+        model_transform_step = TransformStep(
+            name=step_name,
+            transformer=transform,
+            inputs=TransformInput(
+                data=step_process.properties.ProcessingOutputConfig.Outputs["transform"].S3Output.S3Uri, 
+                content_type='text/csv'
+            )
+        )
+        logger.info('Transform step created')
+        return model_transform_step
+
+    def get_data_quality_step(self, step_process)->ClarifyCheckStep:
+        ''' 
+        create a data quality step to be used in SageMaker pipeline. The data quality step will compare 
+        data distribution with the baseline, which is generated during training
+
+        Args:
+            step_process: data quality step depends on the process step
+
+        Returns:    
+            ClarifyCheckStep
+        
+        '''
+        from deploy_endpoint.get_approved_package import get_approved_package_desc
+        spec = get_approved_package_desc()
+        processed_s3_uri = step_process.properties.ProcessingOutputConfig.Outputs["transform"].S3Output.S3Uri
+
+        step_name = "data_quality"
+        output_path = self.get_output_path(step_name)
+
+        data_quality_check_config = DataQualityCheckConfig(
+            baseline_dataset=processed_s3_uri,
+            dataset_format=DatasetFormat.csv(header=False, output_columns_position="START"),
+            output_s3_uri=output_path
+        )
+
+        data_quality_check_step = QualityCheckStep(
+            name="CheckDataQuality",
+            skip_check=False,
+            register_new_baseline=False,
+            quality_check_config=data_quality_check_config,
+            check_job_config=self.check_job_config,
+            supplied_baseline_statistics=spec['DriftCheckBaselines']['ModelDataQuality']['Statistics']['S3Uri'],
+            supplied_baseline_constraints=spec['DriftCheckBaselines']['ModelDataQuality']['Constraints']['S3Uri'],
+            # model_package_group_name=spec['ModelPackageGroupName'] #this fails in staging & prod as model package in Dev only
+        )      
+
+        return data_quality_check_step  
+
+
+    def get_pipeline(self)->Pipeline:
+        ''' 
+        create a SageMaker pipeline to be used in SageMaker pipeline
+        '''
+        
+        step_process = self.get_process_step()
+        step_data_quality = self.get_data_quality_step(step_process)
```

Comment:
The data quality baseline calculation step is not in the training pipeline, so the data quality check in the inference pipeline fails. At what point is the baseline calculated?